core/modules/file/file.entity.inc

```diff
@@ -126,6 +126,77 @@ class File extends Entity {
       'options' => array(),
     );
   }
+
+  /**
+   * Overrides Entity::createAccess().
+   */
+  public static function createAccess($bundle = NULL, $account = NULL) {
+    $rights = &backdrop_static(__METHOD__, array());
+
+    // If no user object is supplied, the access check is for the current user.
+    if (empty($account)) {
+      $account = $GLOBALS['user'];
+    }
+
+    // If we've already checked access for this node, user and op, return from
+    // cache.
+    if (isset($rights[$account->uid][$bundle])) {
+      return $rights[$account->uid][$bundle];
+    }
+
+    $rights[$account->uid][$bundle] = user_access('access content', $account);
```

Comment:
Is "access content" the right permission here? Seems like if we don't yet have a create file permission, this should just return TRUE all the time? Or NULL?

Reply:
I grabbed this from file.module
See

$items['file/ajax'] = array(
    'page callback' => 'file_ajax_upload',
    'delivery callback' => 'ajax_deliver',
    'access arguments' => array('access content'),
    'theme callback' => 'ajax_base_page_theme',
    'type' => MENU_CALLBACK,
  );

I can change it to TRUE, but it seems impossible to upload a file without 'access content' permission.

Reply:
Sounds fair to me. This permission is extremely rarely disabled anyway.